internal/node/launcher.sh

```diff
@@ -291,6 +271,33 @@ if [[ -n "${BAZEL_NODE_MODULES_ROOT:-}" ]]; then
   fi
 fi
 
+if [ "$PATCH_REQUIRE" = true ]; then
+  require_patch_script=${BAZEL_NODE_PATCH_REQUIRE}
+  # See comment above
+  case "${require_patch_script}" in
+    # Absolute path on unix
+    /*          ) ;;
+    # Absolute path on Windows, e.g. C:/path/to/thing
+    [a-zA-Z]:/* ) ;;
+    # Otherwise it needs to be made relative
+    *           ) require_patch_script="./${require_patch_script}" ;;
+  esac
+  LAUNCHER_NODE_OPTIONS+=( "--require" "$require_patch_script" )
+  # Change the entry point to be the loader.js script so we run code before node
+  MAIN=$(rlocation "TEMPLATED_loader_script")
+else
+  # Entry point is the user-supplied script
+  MAIN=TEMPLATED_entry_point_execroot_path  
+  # TODO: after we link-all-bins we should not need this extra lookup
+  if [[ ! -f "$MAIN" ]]; then
+    if [ "$FROM_EXECROOT" = true ]; then
+      MAIN="$EXECROOT/$MAIN"
```

Comment:
Tests seem to pass except for `//internal/node/test:main_test` when executed with `bazel test`; it works fine with `bazel run`. Maybe this is because paths change when running in the sandbox? The code could probably be adapted to also deal with that case

Reply:
okay I think we should immediately attempt to get the default flipped in a PR following this one.

reasoning:
- will have to wait 6mo otherwise
- long flag flip rollouts are a bummer
- in the meantime if other users hit the same bug you did it will be really hard for them to discover this opt-in flag that fixes it